### DISCUSSION TITLE
VLV control search issues

### DISCUSSION THREAD

**Discussion by ParsedID** [dsbenghe/Novell.Directory.Ldap.NETStandard#248] [General]
There are some issues we found with the VLV control in our testing.

An unexpected result occurs when you have full page sizes.
Example scenario: 
There are n entries that are returned by the query and page size is set to any factor of n.
An example would be a query that returns 200 items and page size is set to 100.
You get an additional page response with only the last entry from the previous page - so a duplicate of the last entry is returned.

The additional issue is that if page size is set to 1 you get an infinite loop since the page is always full and the VLV control handler always requests a new page.

Now as we tested, this seems to be a server side behaviour, since if you set the startIndex of LdapVirtualListControl to anything higher than the number of results, the server always returns the last entry.

The only idea i got so far is to ensure an attribute that is always present on all entries would be included for VLV (perhaps objectGUID)  in the handler and to check the existing result set for any duplicate and cease the operation once one is encountered.

If this sounds like an acceptable solution i can make a PR for it?
Or is there a better solution that i just fail to see?

**Comment by ghost** [comment]
I'd check to see if your startIndex (one based) and afterCounts are correct.
VLV controls works for me (though it seems non-intuitive at first).

Check out this book, its helpful: [Understanding and Deploying LDAP Directory Services](https://books.google.it/books?id=fcW1xl1BejUC&pg=PA135&lpg=PA135&dq=LdapVirtualListControl%20how%20to%20use&source=bl&ots=b3cjiaTFoK&sig=1_iCXxkqE_c4wexa_bUmXssGNO0&hl=it&sa=X&ved=2ahUKEwib4fDYk7jcAhUJ36QKHf3gChYQ6AEwBnoECAUQAQ#v=onepage&q=LdapVirtualListControl%20how%20to%20use&f=false)
